### PR TITLE
Fix snyk alerts

### PR DESCRIPTION
**Pull request by danielhoherd**
**Which issue this PR fixes**:

- Bump some versions to fix snyk alerts
- Add pre-commit from ap-base
- Run `pre-commit run --all-files` which auto-fixed a bunch of whitespace stuff.
- Fix a bunch of `shellcheck` complaints
- Update a circleci generator from ap-base

**Summary of changes**:

**Which images are updated by this PR?**:

- [ ] alertmanager
- [ ] blackbox-exporter
- [ ] configmap-reloader 
- [ ] curator
- [ ] elasticsearch
- [ ] elasticsearch-exporter
- [ ] fluentd
- [x] grafana
- [ ] keda
- [ ] keda-metrics-adapter
- [ ] kibana
- [ ] kube-state
- [ ] kubed
- [ ] nats-server
- [ ] nats-streaming
- [ ] nginx
- [x] nginx-es
- [ ] node-exporter
- [ ] pgbouncer
- [ ] pgbouncer-exporter
- [x] postgres-exporter
- [x] prisma
- [x] prometheus
- [ ] redis
- [ ] registry
- [ ] statsd-exporter

**Checklist** (required)

Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.

- [x] version.txt is updated in the changed image(s)

If a security scan fails for an unchanged image...

- [ ]  a fix has been applied OR...
- [ ]  an [issue]() has been created

If adding a new image:

- [ ] the directory has the same name you intend it to be published as, less a preceding "ap-"
- [ ] the directory includes a file "version.txt", with version matching the underlying software version
- [ ] the directory includes a file "cve-whitelist.yaml", which may be empty
- [ ] the file .github/PULL_REQUEST_TEMPLATE.md is updated to include the image in the checklist
- [ ] execute the script .circleci/generate_circleci_config.py, commit changes to .circleci/config.yml

**Additional notes for your reviewer**:

### CODE REVIEW

File: curator/Dockerfile [hunk 1]
```diff
@@ -13,15 +13,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM astronomerinc/ap-base:3.12-1
+FROM quay.io/astronomer/ap-base:3.12.2
```
Comment (vishwas-astro) [inline curator/Dockerfile:16]: Do you want to use latest ap-base:3.12.3 instead?

File: nginx-es/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM astronomerinc/ap-base:3.12-1
+FROM quay.io/astronomer/ap-base:3.12.2
```
Comment (vishwas-astro) [inline nginx-es/Dockerfile:16]: again, can we not use ap-base:3.12.3?

File: postgres-exporter/Dockerfile [hunk 3]
```diff
@@ -20,8 +20,8 @@ FROM wrouesnel/postgres_exporter:v0.8.0 AS upstream
 # I would like some normal tooling like package manager and shell available.
 # In the next update, I think they are migrating from the scratch image,
 # so it will likely be unnecessary to perform this multi-stage build.
-# https://hub.docker.com/r/astronomerinc/ap-base/
-FROM astronomerinc/ap-base:3.12-1
+# https://quay.io/repository/astronomer/ap-base
+FROM quay.io/astronomer/ap-base:3.12.2
```
Comment (vishwas-astro) [inline postgres-exporter/Dockerfile:24]: can we use ap-base:3.12.3?

**Comment by jpweber** [comment]
Since there is a ap-base:3.12.3 I will echo vishwa's mention of using that base for the things we've touched here.

**Review by jpweber — CHANGES_REQUESTED** [review]
Since there is a ap-base:3.12.3 I will echo vishwa's mention of using that base for the things we've touched here.

**Review by jpweber — APPROVED** [review]

**Review by jpweber — APPROVED** [review]
ooh. Nice touch.

**Review by vishwas-astro — APPROVED** [review]

**Pull request merged**